architectures/decentralized/testing/src/docker_setup.rs

```diff
@@ -333,3 +435,96 @@ pub async fn kill_all_clients(docker: &Docker, signal: &str) {
     // Small delay to ensure containers terminate
     tokio::time::sleep(Duration::from_secs(2)).await;
 }
+
+/// Extract a Solana keypair from a container and save to host filesystem
+pub async fn extract_keypair_from_container(
+    docker: &Docker,
+    container_name: &str,
+    output_path: &str,
+) -> Result<(), String> {
+    use bollard::container::DownloadFromContainerOptions;
+    use std::fs;
+    use std::io::Write;
+    use tar::Archive;
+
+    let keypair_container_path = "/root/.config/solana/id.json";
+    let download_options = Some(DownloadFromContainerOptions {
+        path: keypair_container_path.to_string(),
+    });
+
+    let mut keypair_tar_stream = docker.download_from_container(container_name, download_options);
+    let mut keypair_tar_bytes = Vec::new();
+
+    while let Some(chunk) = keypair_tar_stream.next().await {
+        keypair_tar_bytes.extend_from_slice(
+            &chunk.map_err(|e| format!("Failed to download keypair: {}", e))?[..],
+        );
+    }
+
+    // Extract from tar archive
+    let mut archive = Archive::new(&keypair_tar_bytes[..]);
+    let mut keypair_json = String::new();
+    if let Some(entry) = archive
+        .entries()
+        .map_err(|e| format!("Failed to read tar: {}", e))?
+        .next()
+    {
+        let mut entry = entry.map_err(|e| format!("Failed to read entry: {}", e))?;
+        use std::io::Read;
+        entry
+            .read_to_string(&mut keypair_json)
+            .map_err(|e| format!("Failed to read keypair: {}", e))?;
+    }
+
+    // Save to host filesystem
+    let mut file = fs::File::create(output_path)
+        .map_err(|e| format!("Failed to create file {}: {}", output_path, e))?;
+    file.write_all(keypair_json.as_bytes())
+        .map_err(|e| format!("Failed to write keypair: {}", e))?;
+
+    println!(
+        "Extracted keypair from {} to {}",
+        container_name, output_path
+    );
+    Ok(())
+}
```

Comment:
instead of extracting the key from a container, can't we just use a pre-defined one and use it when spawning the docker instance?